cascade-merge/action.yml

```diff
@@ -0,0 +1,213 @@
+#                  }
+#                ]
+#              },
+#              {
+#                "type": "divider"
+#              }
+#            ]
+#          }
+#
+
+    # checkout again to clean all local merges done in previous step
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0  # fetch all history so we can push to other branches
+
+    - name: Cascade merge
+      if: "! contains(env.COMMIT_MSG, '[skip cascade merge]')"  # continue only if the message does not contain '[skip cascade merge]'
+      run: |
+        # Set Bash to print all commands and stop executing on any error
+        # set -ex
+
+        # Define helper functions to echo messages in color
+        function echo_green {
+            echo -e "\e[32m$1\e[0m"
+            echo $1 >> $GITHUB_STEP_SUMMARY
+        }
+        function echo_red {
+            echo -e "\e[31m$1\e[0m"
+            echo $1 >> $GITHUB_STEP_SUMMARY
+        }
+        
+        # Define function to handle PR creation
+        function handle_pr_creation {
+          echo_green "Creating PR to cascade merge from $1 to $2"
+          NEW_PR_URL=$(gh pr create --title "Cascade merge from $1 to $2" --body "Automated cascade merge" --base $2 --head $1 --repo ${{ github.repository }})
+          PR_CREATION_STATUS=$?
+
+          if [[ $PR_CREATION_STATUS -ne 0 ]]; then
+            echo_red "PR creation failed."
+            exit 1
+          else
+            echo_green "PR is created and ready for review: $NEW_PR_URL"
+            exit 0
+          fi
+        }
+        
+        # Constants
+        BACKPORT_TARGET_BRANCH_PATTERN="release/"
+        
+        # Extract original branch and version
+        ORIGINAL_BRANCH="${GITHUB_REF#refs/heads/}"
+        ORIGINAL_VERSION="${ORIGINAL_BRANCH#"$BACKPORT_TARGET_BRANCH_PATTERN"}"
+        
+        # Get release versions from remote branches
+        RELEASE_VERSIONS=$(git branch -r | grep -oP "origin/${BACKPORT_TARGET_BRANCH_PATTERN}\\K.*" | sort -Vr)
+        CURRENT_INDEX=$(echo $RELEASE_VERSIONS | tr ' ' '\n' | grep -n $ORIGINAL_VERSION | cut -d: -f1)
+
+
+        # Loop through each version in the release versions list before the current index
+        for TARGET_VERSION in $(echo $RELEASE_VERSIONS | tr ' ' '\n' | head -n $((CURRENT_INDEX - 1))); do
+          TARGET_BRANCH="$BACKPORT_TARGET_BRANCH_PATTERN$TARGET_VERSION"
+          
+          # Create a local branch that tracks the remote branch
+          git branch $TARGET_BRANCH origin/$TARGET_BRANCH
+        
+          # Attempt to merge the original branch into the target branch
+          git checkout $TARGET_BRANCH
+          git merge $ORIGINAL_BRANCH --no-commit --no-ff
+        
+          # Check if the merge was successful
+          if [ $? -ne 0 ]; then
+            echo_red "Merge conflict detected. Skipping branch $TARGET_BRANCH."
```

Comment:
all the conflct detection part was already done earlier, I think you can simplify code

Reply:
refactored in a pull request https://github.com/LedgerHQ/actions/pull/86